core/bin/qdrant/create_collection.rs

```diff
@@ -0,0 +1,146 @@
+async fn create_qdrant_collection(
+    cluster: QdrantCluster,
+    provider_id: ProviderID,
+    model_id: SupportedEmbedderModels,
+) -> Result<()> {
+    let qdrant_clients = QdrantClients::build().await?;
+    let client = qdrant_clients.client(cluster);
+    let raw_client = client.raw_client();
+
+    let embedder = provider(provider_id).embedder(model_id.to_string());
+
+    let collection_name = format!(
+        "{}_{}_{}",
+        client.collection_prefix(),
+        provider_id,
+        model_id
+    );
+
+    println!(
+        "About to create collection {} on cluster {}",
+        collection_name, cluster
+    );
+
+    match utils::confirm(&format!(
+        "Are you sure you want to create collection {} on cluster {}?",
+        collection_name, cluster
+    ))? {
+        true => (),
+        false => Err(anyhow!("Aborted"))?,
+    }
+
+    // See https://www.notion.so/dust-tt/Design-Doc-Qdrant-re-arch-d0ebdd6ae8244ff593cdf10f08988c27.
+
+    // First, we create the collection.
+    let res = raw_client
+        .create_collection(&qdrant::CreateCollection {
+            collection_name: collection_name.clone(),
+            vectors_config: Some(qdrant::VectorsConfig {
+                config: Some(qdrant::vectors_config::Config::Params(
+                    qdrant::VectorParams {
+                        size: embedder.embedding_size() as u64,
+                        distance: qdrant::Distance::Cosine.into(),
+                        on_disk: Some(true),
+                        ..Default::default()
+                    },
+                )),
+            }),
+            hnsw_config: Some(qdrant::HnswConfigDiff {
+                payload_m: Some(16),
+                m: Some(0),
+                ..Default::default()
+            }),
+            optimizers_config: Some(qdrant::OptimizersConfigDiff {
+                memmap_threshold: Some(16384),
+                ..Default::default()
+            }),
+            quantization_config: Some(qdrant::QuantizationConfig {
+                quantization: Some(qdrant::quantization_config::Quantization::Scalar(
+                    qdrant::ScalarQuantization {
+                        r#type: qdrant::QuantizationType::Int8.into(),
+                        quantile: Some(0.99),
+                        always_ram: Some(true),
+                    },
+                )),
+            }),
+            on_disk_payload: Some(true),
+            sharding_method: Some(qdrant::ShardingMethod::Custom.into()),
+            shard_number: Some(2),
+            replication_factor: Some(2),
+            write_consistency_factor: Some(1),
+            ..Default::default()
+        })
+        .await?;
+
+    // Then, we create the 24 shard_keys.
+    for i in 0..SHARD_KEY_COUNT {
+        raw_client
+            .create_shard_key(
+                collection_name.clone(),
+                &qdrant::shard_key::Key::Keyword(format!("{}_{}", client.shard_key_prefix(), i)),
+                Some(2),
```

Comment:
Big question is why do we need to repeat these configs at shard creation.

Mind asking them on slack?

Reply:
Asked [here](https://dust4ai.slack.com/archives/C05HBLG1TRB/p1716449483486159).

TL;DR: They are not required when creating a shard key.